src/api/creatures.py

```diff
@@ -288,91 +319,97 @@ def play_with_creature(user_id: int, creature_id: int):
     Playing with a pet at max happiness does not earn the user any gold or affinity.
     """
 
-    played = False
-    gold_earned = 0
-    affinity_change = 0
-    happiness_change = 0
-
     try:
         with db.engine.begin() as connection:
             stats = (
                 connection.execute(
                     sqlalchemy.text(
                         """
-                    SELECT 
-                        happiness, e.max_happiness,
-                        COALESCE(user_creature_connection.affinity, 0) AS affinity
-                    FROM creatures 
-                    JOIN creature_types 
-                        ON creatures.type = creature_types.type AND creatures.id = :creature
-                    JOIN evolution_stages e
-                        ON creatures.stage = e.stage
-                    LEFT JOIN user_creature_connection 
-                        ON creatures.id = user_creature_connection.creature_id AND user_id = :user_id
-                """
+                        SELECT
+                            name,
+                            happiness,
+                            e.max_happiness
+                        FROM creatures
+                        JOIN creature_types
+                            ON creature_types.type = creatures.type
+                            AND creatures.id = :creature
+                        JOIN evolution_stages e
+                            ON creatures.stage = e.stage
+                        JOIN users
+                            ON users.id = :user_id
+                        """
                     ),
                     {"creature": creature_id, "user_id": user_id},
                 )
                 .mappings()
-                .fetchone()
+                .one_or_none()
             )
+            if not stats:
+                raise HTTPException(
+                    status_code=404,
+                    detail=f"User {user_id} and/or creature {creature_id} does not exist.",
+                )
 
-            print(f"Stats: {stats}")
+            print(
+                f"[play_with_creature] Creature {creature_id} info for user {user_id}:",
+                stats,
+            )
 
-            if stats["happiness"] <= stats["max_happiness"] - 1:
-                happiness_change = 1
-                played = True
-                gold_earned = 2
-                affinity_change = 1
+            if stats["happiness"] == stats["max_happiness"]:
```

Comment:
I would change this to ">=" instead of "==" in case there is some issue or bug that causes "happiness" to be greater than the max